### PR TITLE
Update missing references to DI

### PR DESCRIPTION
**Pull request by MrPetovan**
Follow-up to #8019 

Please merge https://github.com/friendica/friendica-addons/pull/933 as well.

### CODE REVIEW

**Comment by MrPetovan** [comment]
Self-merging because the current instability of the codebase.

File: mod/redir.php [hunk 1]
```diff
@@ -151,7 +151,7 @@ function redir_magic($a, $cid, $url)
 	// We don't use magic auth when there is no visitor, we are on the same system or we visit our own stuff
 	if (empty($visitor) || Strings::compareLink($basepath, System::baseUrl()) || Strings::compareLink($contact_url, $visitor)) {
 		Logger::info('Redirecting without magic', ['target' => $target_url, 'visitor' => $visitor, 'contact' => $contact_url]);
-		$a->redirect($target_url);
+		System::externalRedirect($target_url);
```
Comment (annando) [inline mod/redir.php]: Why not `DI::baseUrl()->redirect();` here?

Comment (MrPetovan) [reply mod/redir.php]: All `DI::baseUrl()->redirect();` does is prepending the base URL to a relative path and then calling `System::externalRedirect`. If we know we already have an external URL we can skip the BaseURL part.

Comment (annando) [reply mod/redir.php]: I just checked the function. I thought that we had a check that the path had to be external only. But this isn't the case (anymore). So it is okay.

**Pull request merged**